### DISCUSSION TITLE
-R, --reverse

### DISCUSSION THREAD

**Discussion by infinitynet2011** [esnet/iperf#1546] [General]
C:\iperf3.14_64>iperf3.exe -c ****  -u -b 100m  -R
Connecting to host *****
Reverse mode, remote host ***** is sending
[  5] local 192.168.0.135 port 65231 connected to **** port 8000
[ ID] Interval           Transfer     Bitrate         Jitter    Lost/Total Datagrams
[  5]   0.00-1.00   sec  11.9 MBytes  99.8 Mbits/sec  0.135 ms  124/8668 (1.4%)
[  5]   1.00-2.00   sec  11.7 MBytes  97.9 Mbits/sec  0.117 ms  178/8562 (2.1%)
[  5]   2.00-3.00   sec  11.7 MBytes  98.0 Mbits/sec  0.080 ms  174/8562 (2%)
[  5]   3.00-4.00   sec  11.6 MBytes  97.3 Mbits/sec  0.084 ms  230/8560 (2.7%)
[  5]   4.00-5.00   sec  11.7 MBytes  97.8 Mbits/sec  0.121 ms  194/8563 (2.3%)
[  5]   5.00-6.00   sec  11.9 MBytes  99.7 Mbits/sec  0.084 ms  21/8561 (0.25%)
[  5]   6.00-7.00   sec  11.7 MBytes  98.0 Mbits/sec  0.107 ms  172/8562 (2%)
[  5]   7.00-8.00   sec  11.5 MBytes  96.1 Mbits/sec  0.671 ms  304/8535 (3.6%)
[  5]   8.00-9.00   sec  11.6 MBytes  97.5 Mbits/sec  0.113 ms  239/8589 (2.8%)
[  5]   9.00-10.00  sec  11.7 MBytes  97.9 Mbits/sec  0.126 ms  179/8561 (2.1%)
[  5]  10.00-10.91  sec  10.7 MBytes  98.7 Mbits/sec  0.131 ms  100/7799 (1.3%)

Without -R, have 0.0% loss on both ends.. its normal?

On the old version of iPerf, I don`t have such problems.

Thx,
Gabe

**Comment by davidBar-On** [comment]
Gabe, UDP packets lost may be different in each direction, since the performance of the client and servers machines and interfaces can be different. However, there should not be any difference when running new or old iperf3 versions.  Few questions:
1. What is the throughput of the interface between the client and server?  You can determine the approximate performance by using TCP test without bitrate limitation.  Running two TCP tests, with and without `-R`, may help to understand whether the throughput in both directions is the same.
2. Which version is the old iperf3 version you used (`iperf3 -v`)?
3. What are the sizes of the UDP packets when running the old and the new versions?  Can be determined when using the `-V` option.
